code/game/objects/items/melee/misc.dm

```diff
@@ -74,11 +74,37 @@
 	custom_materials = list(/datum/material/iron = 1000)
 	wound_bonus = 10
 	bare_wound_bonus = 25
+	var/lunge_delay = 1.2 SECONDS
 
 /obj/item/melee/sabre/Initialize()
 	. = ..()
 	AddComponent(/datum/component/butchering, 30, 95, 5) //fast and effective, but as a sword, it might damage the results.
 
+/obj/item/melee/sabre/examine(datum/source, mob/user, list/examine_list)
+	. = ..()
+	. += "<span class='notice'><b>right-click</b> to lunge with [src].</span>"
+
+/obj/item/melee/sabre/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
+	if(user.Adjacent(target))
+		return
```

Comment:
why does this need you to be next to your target

Reply:
This specifically returns when you're next to your target, though it is a bit weird you cant just lunge from adjacency anyway, since it's a 1.20 second do_after they can break with a shove or bumping into you.